reasoner/resolution/resolver/SubsumptiveCoordinator.java

```diff
@@ -39,21 +40,30 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(SubsumptiveCoordinator.class);
     private final Map<AnswerState.Partial<?>, Request.Factory> requestFactories;
+    protected final Set<Mapping> equivalentMappings;
+    protected final Map<ConceptMap, Mapping> reflexiveMappings;
     protected boolean isInitialised;
 
-    public SubsumptiveCoordinator(Driver<RESOLVER> driver, String name, ResolverRegistry registry) {
+    protected SubsumptiveCoordinator(Driver<RESOLVER> driver, String name, Set<Mapping> equivalentMappings,
+                                     ResolverRegistry registry) {
         super(driver, name, registry);
         this.isInitialised = false;
         this.requestFactories = new HashMap<>();
+        this.equivalentMappings = equivalentMappings;
+        this.reflexiveMappings = new HashMap<>();
     }
 
     @Override
     public void receiveVisit(Request.Visit fromUpstream) {
         LOG.trace("{}: received Visit: {}", name(), fromUpstream);
         if (!isInitialised) initialiseDownstreamResolvers();
         if (isTerminated()) return;
-        Driver<? extends Resolver<?>> worker = getOrCreateBoundResolver(fromUpstream.partialAnswer());
-        Request.Factory requestFactory = getOrCreateRequestFactory(fromUpstream.partialAnswer(), worker);
+        ConceptMap conceptMap = fromUpstream.partialAnswer().conceptMap();
+        Mapping mapping = reflexiveMappings.get(conceptMap);
+        if (mapping == null) mapping = computeReflexiveMappings(conceptMap);
```

Comment:
the story abstraction level is:
`ConceptMap` -> `Mapping` -> Worker
the implementation detail of how to compute the mapping can be pushed one layer down - in the future we could do a more or less efficient implementation, but the story stays the same.

Siblings should stay at the same level of abstraction when reading the high level story.